static/js/components/material/Dialog.js

```diff
@@ -45,13 +47,14 @@ export default class Dialog extends React.Component {
   }
 
   componentWillReceiveProps(nextProps: DialogProps) {
-    if (this.dialog) {
-      if (this.props.open !== nextProps.open) {
-        if (nextProps.open) {
+    if (this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.createDialog();
+        if (this.dialog) {
           this.dialog.show();
-        } else {
-          this.dialog.close();
         }
+      } else if (this.dialog) {
+        this.dialog.destroy();
       }
     }
   }
```

Comment:
I tinkered with this a little locally to see how the MDCDialog object behaves. We can make this a bit more concise with something like this:

```javascript
  showMdc() {
    if (this.dialog) {
      this.dialog.show();
    }
  }

  destroyMdc() {
    if (this.dialog) {
      this.dialog.destroy();
    }
  }

  componentDidMount() {
    this.createDialog();
    if (this.props.show) this.showMdc();
  }

  componentWillUnmount() {
    this.destroyMdc();
  }

  componentWillReceiveProps(nextProps: DialogProps) {
    if (this.props.open !== nextProps.open) {
      if (nextProps.open) {
        this.showMdc();
      } else {
        this.destroyMdc();
      }
    }
  }
```

Reply:
OK